runtime/tests/checker/attachments_test.go

```diff
@@ -0,0 +1,1486 @@
+			}
+			attachment Test for R: I {
+				let x: Int
+				init(x: Int) {
+					self.x = x
+				}
+			}`,
+		)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("method", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			resource R {}
+			resource interface I {
+				fun x(): Int
+			}
+			attachment Test for R: I {
+				fun x(): Int { return 0 }
+			}`,
+		)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("field missing", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			resource R {}
+			resource interface I {
+				let x: Int
+			}
+			attachment Test for R: I {
+			}`,
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		assert.IsType(t, &sema.ConformanceError{}, errs[0])
+	})
+
+	t.Run("field type", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			resource R {}
+			resource interface I {
+				let x: Int
+			}
+			attachment Test for R: I {
+				let x: AnyStruct
+				init(x: AnyStruct) {
+					self.x = x
+				}
+			}`,
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		assert.IsType(t, &sema.ConformanceError{}, errs[0])
+	})
+
+	t.Run("method missing", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			resource R {}
+			resource interface I {
+				fun x(): Int
+			}
+			attachment Test for R: I {
+			}`,
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		assert.IsType(t, &sema.ConformanceError{}, errs[0])
+	})
+
+	t.Run("method type", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			resource R {}
+			resource interface I {
+				fun x(): Int
+			}
+			attachment Test for R: I {
+				fun x(): AnyStruct { return "" }
+			}`,
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		assert.IsType(t, &sema.ConformanceError{}, errs[0])
+	})
+
+	t.Run("method missing, exists in base type", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			resource R {
+				fun x(): Int { return 3 }
+			}
+			resource interface I {
+				fun x(): Int
+			}
+			attachment Test for R: I {
+			}`,
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		assert.IsType(t, &sema.ConformanceError{}, errs[0])
+	})
+
+	t.Run("kind mismatch resource", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			resource R {}
+			struct interface I {}
+			attachment Test for R: I {}`,
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		assert.IsType(t, &sema.CompositeKindMismatchError{}, errs[0])
+	})
+
+	t.Run("kind mismatch struct", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			struct R {}
+			resource interface I {}
+			attachment Test for R: I {}`,
+		)
+
+		errs := RequireCheckerErrors(t, err, 1)
+
+		assert.IsType(t, &sema.CompositeKindMismatchError{}, errs[0])
+	})
+
+	t.Run("conforms to base", func(t *testing.T) {
+
+		t.Parallel()
+
+		_, err := ParseAndCheck(t,
+			`
+			resource interface I {}
+			attachment A for I: I {}`,
```

Comment:
This sounds like should be an error. Interfaces cannot conform to interfaces (yet).
Even when we add that, self-conforming would/should be rejected.

Reply:
This isn't saying that `I` conforms to `I`, it's saying that `A` conforms to `I` and also has a base of type `I`. So for example:
```
resource interface I {
    fun foo() {}
}
attachment A for I: I {
    fun foo() {
        base.foo() // delegate implementation to base
    }
}
```

Reply:
oh okay, I misread the code there